### ISSUE TITLE
Make sure colorings schema works for multi-colorby

### ISSUE THREAD

**Issue by trvrb** [nextstrain/augur#184]
We would like to allow multi-colorbys or matrix colorbys like exist in Phandango and elsewhere. Would be particularly useful for flu HI, where columns would be individual sera.

Currently, `zika.json` contains `colorings` that look like:
```
 "colorings": {
    "region": {
      "type": "categorical",
      "title": "region",
      "scale": {
        "oceania": "#88BB6C",
        "south_america": "#E56C2F",
        "north_america": "#DC2F24"
      }
    },
    "sequence": {
      "type": "categorical",
      "title": "sequence"
    },
    "gt": {
      "title": "Genotype",
      "type": "ordinal"
    },
    "num_date": {
      "title": "Sampling date",
      "type": "continuous"
    }
  },
```

where things like `region` map to a particular `trait` entry of a tree node. I would propose to keep this basic structure for multi-colorbys. There would be a `trait` entry of each tree node that looks like:
```
 "traits": {
  "sera": {
    "value": [0,4,2,3,null,1,4],
  },
```
so that multi-colorby traits have values that are arrays. I would think that
```
"colorings": {
  "sera": {
    "title": "Serum titers",
    "type": "continuous"
  }
}
```
should then work for this.

So, I believe current proposal works for multi-colorbys, but I wanted to be sure. @jameshadfield, do you see a problem with doing multi-colorbys with current schema v2?

**Comment by jameshadfield** [comment]
So auspice would detect that the value is `array` and instead of colouring the tree it would produce phandango-like columns...
Would there be a way to display multiple (normal) colorBys in this style?

**Comment by trvrb** [comment]
I see... I definitely agree with the desire to do things like have a `geo` multi-coloby that shows columns for `division`, `country` and `region`. Current thoughts on how to accomplish this would be something like:
```
  "color_groups": {
    "geo": {
      "title": "Geography",
      "tree": "country",
      "matrix": ["country", "region"]
    },
    "sera": {
      "title": "Antigenic phenotypes",
      "tree": "cHI",
      "matrix": ["WI05", "CA07", "BR09", "VI11"]
    }
  },
  "colorings": {
    "region": {
      "type": "categorical",
      "title": "region",
      "scale": {
        "oceania": "#88BB6C",
        "south_america": "#E56C2F",
        "north_america": "#DC2F24"
      }
    },
    "country": {
      "type": "categorical",
      "title": "country",
    },
    "cHI": {
      "title": "Antigenic advancement",
      "type": "continuous"
    },
    "WI05": {
      "display": false,
      "type": "continuous"
    },
    "CA07": {
      "display": false,
      "type": "continuous"
    },
    "BR09": {
      "display": false,
      "type": "continuous"
    },
    "VI11": {
      "display": false,
      "type": "continuous"
    }
  },
```

`colorings` stays largely the same. The only addition is `"display": false` that makes a coloring not appear in the dropdown, eg we don't want 50 different sera colorings cluttering up the dropdown. But we have the addition of `color_groups` that points at keys in `colorings`. Can separately specify how to color tree and how to construct matrix. Each column in a matrix, like `region` would take it's colors directly from `colorings`.

**Comment by jameshadfield** [comment]
Since this proposal is a backwards-compatable extension to the auspice v2 JSONs, it can be mostly applied after release. I think I'll enable the `"display": false` option now as it's rather non-controversial, and then we can deal with the `"color_groups"` later. I imagine when we implement `"color_groups"` we may also want to design an auspice UI for selecting multiple single color-bys.

**Comment by trvrb** [comment]
If go with the proposed `color_groups` plus `display` as described above then this can wait until schema v2.1 (or whatever). It should be fully backward compatible.

I'm okay axing this from initial v6 release.

**Comment by emmahodcroft** [comment]
I think I'd vote to leave this from v6 for now, because I think it will be worth really having a deep think about all situations we want to enable, and best to do that when we have more time (wouldn't want this to hold up v6). I'm pretty excited about getting this implemented though!

James, would it be worth adding a 'future' tag (or similar, maybe 'v7', or 'post-v6'?) to help us keep track of these issues that we want soon but are cutting from v6? So that it's easy to come back to them later! Or if you don't think this is a problem (or there's a better way to do this), ignore!

**Comment by trvrb** [comment]
I don't think this needs to be dealt with any time soon, but I'd like to leave the issue open as prior art for thinking about multi-colorby implementation. Shelving in "icebox".
